.github/workflows/livetest.yml

```diff
@@ -0,0 +1,149 @@
+        # WARNING: if you want to add this users to sudoers make sure to propertly delete the SINK_SSHRELAY_PRIVATE
+        run: sudo useradd --create-home livetest; echo livetest:foobar | sudo chpasswd
+
+      - name: Download dists tarball for current git SHA from GitHub
+        id: make_dist
+        continue-on-error: true
+        run: ./test/make_dist.py -d
+
+      - name: Write a comment on the PR prompting the user to try later if make_dist.py failed
+        uses: actions/github-script@v3
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            github.issues.createComment({
+              issue_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              body: 'Download of the dists artifact failed: Retry again in 10 minutes'
+            })
+        if: steps.make_dist.outcome == 'failure'
+
+      - run: exit 1
+        if: steps.make_dist.outcome == 'failure'
+
+      - name: Install dependencies
+        run: |
+          sudo sed -Ei 's/^# deb-src /deb-src /' /etc/apt/sources.list
+          sudo apt update
+          sudo apt-get -y build-dep cockpit
+          ./tools/make-debs --quick
+          # FIXME: pcp is just utterly slow to start makes the installation fail
+          rm -f cockpit-pcp*.deb
+          for deb in cockpit*.deb; do sudo apt -y install ./$deb; done
+          sudo systemctl start cockpit.socket
+
+      - name: Get the commit SHA that triggered the workflow
+        id: get_commit_sha
+        run: |
+          echo "::set-output name=sha_short::$(git rev-parse --short HEAD)"
+
+      - name: Get ssh gateway port to forward to
+        id: remote_port
+        env:
+          SHA_SHORT: "${{ steps.get_commit_sha.outputs.sha_short }}"
+        run: |
+            echo ::set-output name=SSH_GATEWAY_PORT::$((16#$SHA_SHORT % 32768 + 1500))
+
+      - name: Write a comment on the PR with link to the cockpit instance
+        uses: actions/github-script@v3
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            github.issues.createComment({
+              issue_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              body: 'Cockpit preview: https://${{ env.RELAY_HOST }}:${{ steps.remote_port.outputs.SSH_GATEWAY_PORT }}'
+            })
+
+      - name: Create an SSH private key from the secret
+        run: |
+          sudo touch /run/SINK_SSHRELAY_PRIVATE && sudo chown $(id -u):$(id -g) /run/SINK_SSHRELAY_PRIVATE
+          echo "${{ secrets.SINK_SSHRELAY_PRIVATE }}" > /run/SINK_SSHRELAY_PRIVATE
+          chmod 600 /run/SINK_SSHRELAY_PRIVATE
+          mkdir -p ~/.ssh &&  ssh-keyscan -H ${{ env.RELAY_HOST }} > ~/.ssh/known_hosts
+
+      - name: Keep ssh connection alive
+        run: sudo sed -i -r 's/#ClientAliveInterval.*/ClientAliveInterval 600/' /etc/ssh/sshd_config
```

Comment:
Great job figuring that out! I suppose that should go back to starter-kit and podman?

Reply:
So, this is not really fixed. I think we might need the relevant options on the client. So if I open the cockpit link, it does not really time out, just after the workflow VMs do, 6h?

But if I don't open the link within 20min it seems that somehow the SSH tunnel breaks and then cockpit is not accessible.

Reply:
Maybe it would help to run a shell command that regularly prints something? I'd need to  open that up on the sink, but that would not be a problem.

Reply:
Sure, let's try it, so let's replace 'sleep infinity' with 'while true; echo 'I am alive'; sleep 60; done'

Reply:
Or something similar.